SimulationRuntime/c/simulation/modelinfo.c

```diff
@@ -269,13 +268,52 @@ static void printProfileBlocks(FILE *fout, FILE *plt, const char *plotFormat, co
   }
 }
 
-static void printEquations(FILE *fout, int n, MODEL_DATA_XML *xml) {
+
+static int getVarIdByName(DATA *data, const char* varName)
+{
+  int i;
+  for(i=0;i<data->modelData->nVariablesReal;++i)
+    if (strcmp(varName, data->modelData->realVarsData[i].info.name) == 0)
+      return data->modelData->realVarsData[i].info.id;
+
+  for(i=0;i<data->modelData->nParametersReal;++i)
+    if (strcmp(varName, data->modelData->realParameterData[i].info.name) == 0)
+      return data->modelData->realParameterData[i].info.id;
+
+  for(i=0;i<data->modelData->nVariablesInteger;++i)
+    if (strcmp(varName, data->modelData->integerVarsData[i].info.name) == 0)
+      return data->modelData->integerVarsData[i].info.id;
+
+  for(i=0;i<data->modelData->nParametersInteger;++i)
+    if (strcmp(varName, data->modelData->integerParameterData[i].info.name) == 0)
+      return data->modelData->integerParameterData[i].info.id;
+
+  for(i=0;i<data->modelData->nVariablesBoolean;++i)
+    if (strcmp(varName, data->modelData->integerParameterData[i].info.name) == 0)
+      return data->modelData->integerParameterData[i].info.id;
```

Comment:
@adrpo: copy&paste error: must be `booleanVarsData` instead of `integerParameterData` in these two lines!